gbpservice/nfp/lifecycle_manager/modules/device_lifecycle_manager.py

```diff
@@ -0,0 +1,763 @@
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+from oslo_log import log as logging
+import oslo_messaging as messaging
+
+from gbpservice.nfp.core.main import Event
+from gbpservice.nfp.core.rpc import RpcAgent
+from gbpservice.nfp.common import topics as nsf_topics
+from gbpservice.nfp.db import nfp_db as nfp_db
+from gbpservice.nfp.db import api as nfp_db_api
+from gbpservice.nfp.lib import extension_manager as ext_mgr
+from gbpservice.nfp.lifecycle_manager.openstack import openstack_driver
+#from gbpservice.nfp.lifecycle_manager.compute.drivers import (
+#    nova_driver)
+'''from gbpservice.nfp.lifecycle_manager.drivers import (
+    haproxy_lifecycle_driver)
+from gbpservice.nfp.lifecycle_manager.drivers import (
+    vyos_lifecycle_driver)
+'''
+from neutron.common import rpc as n_rpc
+from neutron import context as n_context
+
+LOG = logging.getLogger(__name__)
+
+
+def rpc_init(controller, config):
```

Comment:
Would it be better to have a base class to make the structure of a module obvious?

class NSFModule():
    def __init__(self, controller, config):
          agent =
          events = 
    def rpc_init():
          self.controller.register_rpc_agents(self.agent)
    def events_init():
          ....

class DLCMModule(NSFModule):
       def __init__(controller, config)
            super().__init__(controler...)
       def rpc_init()
            super().agent = ...
            super().rpc_init()
      def events_init()
            ....